lib/wasix/src/bin_factory/binary_package.rs

```diff
@@ -173,18 +173,25 @@ impl BinaryPackage {
         self.commands.iter().find(|cmd| cmd.name() == name)
     }
 
-    /// Get the bytes for the entrypoint command.
-    pub fn entrypoint_bytes(&self) -> Option<&[u8]> {
+    /// Resolve the entrypoint command name to a [`BinaryPackageCommand`].
+    fn get_entrypoint_cmd(&self) -> Option<&BinaryPackageCommand> {
         self.entrypoint_cmd
             .as_deref()
             .and_then(|name| self.get_command(name))
-            .map(|entry| entry.atom())
     }
 
+    /// Get the bytes for the entrypoint command.
+    pub fn entrypoint_bytes(&self) -> Option<&[u8]> {
+        self.get_entrypoint_cmd().map(|entry| entry.atom())
+    }
```

Comment:
It seems `entrypoint_bytes` is only used in `bin_factor/exec.rs`. We should remove this function and refactor that code that is using it to be simpler:

```rust
if let Some(cmd) = binary.get_command(name).or(binary.get_entrypoint_cmd()) {
   ...
}
else {
}
```

Reply:
It's a public function, we shouldn't just remove those randomly.

Reply:
We can mark it as deprecated if you prefer, and stop using it on our own code.

I don't think it adds any value (as we are the main users of the crate), but is your call if you prefer to deprecate it instead of deleting it

Reply:
Deprecated